src/rpc/misc.cpp

```diff
@@ -364,6 +364,35 @@ static UniValue setmocktime(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+static UniValue runcommand(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+            RPCHelpMan{"runcommand",
+                "\nRun command and parse JSON from stdout (-regtest only)\n",
```

Comment:
Why is this RPC exposed? For testing purposes? It should be a unit test in that case.

Reply:
It's restricted to regtest. Main reasons I didn't build a unit test:
1. no idea how
2. potentially creates even more cross-platform headaches
3. `runCommand` is also only tested via functional tests (via `blocknotify`, etc)

Reply:
> potentially creates even more cross-platform headaches

What you mean? Because of the command itself?

Reply:
@Sjors how about supporting `-exec "cmd"` in bitcoin_test binary (or in any other test binary), then that could be used in the functional test (instead of this RPC).

Reply:
I think it's ok to have this RPC only enabled in regtest. If you wanted to add more protection against misuse, you could disable it without `gArgs.GetBoolArg("-enable-unsafe-runcommand-rpc", false)` or similar.

If you did want to drop the RPC and just write a unit test for `runCommandParseJSON`, I don't think it would be difficult to do by just having the unit test write a temporary json file and then calling `runCommandParseJSON` with `cat path_to_file` on unix and `type path_to_file` on windows.